File: p20.py
def fact(n):
    f = 1
    for i in range(2, n + 1):
        f *= i 
    return f

def sum_digits(n):
    s = 0
    while n:
        s += n % 10
        n //= 10
    return s

def main():
    print(sum_digits(fact(100)))

if __name__ == "__main__":
    main()

File: p20.hs
fact :: Integer -> Integer
fact n
 | n <= 1 = 1
 | otherwise = n * fact (n - 1)

sum_digits :: Integer -> Integer
sum_digits 0 = 0
sum_digits n = (mod n 10) + sum_digits (n `div` 10)

main :: IO()
main = do
    print (sum_digits (fact 100))
